app/controllers/services_controller.rb

```diff
@@ -101,6 +105,13 @@ def service_params
               accessibility_ids: []
           ]
         )
+
+        if result_params['local_offer_attributes']&.[]('survey_answers')
+            result_params['local_offer_attributes']['survey_answers'] =
+                result_params['local_offer_attributes']['survey_answers'].to_h.map{|k,v| { id: k.to_i, answer: v['answer']}}
+        end
+
+        result_params
```

Comment:
can you summarise what this section is doing?

Reply:
This is what the params object looks like when it comes in for the questionnaire:
```
{
"1"=><ActionController::Parameters {"answer"=>"test"} permitted: true>,
 "2"=><ActionController::Parameters {"answer"=>"test2"} permitted: true>,
 "3"=><ActionController::Parameters {"answer"=>"5555"} permitted: true>,
 "4"=><ActionController::Parameters {"answer"=>""} permitted: true>, 
"5"=><ActionController::Parameters {"answer"=>""} permitted: true>,
 "6"=><ActionController::Parameters {"answer"=>""} permitted: true>,
 "7"=><ActionController::Parameters {"answer"=>""} permitted: true>
}
```

This turns it into a JSON array, using the key as the ID field, resulting in a structure like:

```
[
{ id: 1, answer: 'Here is the answer' }
]
```

Reply:
Have added a comment to clarify in the code too